lib/src/Uncertainty/Algorithm/Sensitivity/HSICEstimatorImplementation.cxx

```diff
@@ -256,6 +329,23 @@ Point HSICEstimatorImplementation::getPValuesPermutation() const
   return PValuesPermutation_;
 }
 
+/* Is it safe to compute the weight matrices in parallel? */
+Bool HSICEstimatorImplementation::isWeightMatrixParallel() const
+{
+  return true;
+}
+
+/* Is it safe to compute the permutation p values in parallel? */
+Bool HSICEstimatorImplementation::isPvaluesParallel() const
+{
+  Bool isParallel = true;
+  for (UnsignedInteger i = 0; i < covarianceList_.getSize(); ++i)
+  {
+    isParallel = isParallel && covarianceList_[i].getImplementation()->isParallel();
```

Comment:
`isParallel && covarianceList_[i].getImplementation()->isParallel() && covarianceList_[inputDimension_].getImplementation()->isParallel();` probably
But not sure it is sufficient

Reply:
I guess you mean:
`isParallel && covarianceList_[dim].getImplementation()->isParallel() && covarianceList_[inputDimension_].getImplementation()->isParallel();`.  (this line would replace the entire for cycle)
Indeed; I believe this is a sufficient check, which would eventually allow to perform part of the permutation computations in parallel, and the remaining ones sequentially.